src/main/java/com/microsoft/azure/functions/internal/MiddlewareContext.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.functions.internal;
+
+import com.microsoft.azure.functions.ExecutionContext;
+
+import java.util.Optional;
+
+/**
+ * Middleware Execution Context
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public interface MiddlewareContext extends ExecutionContext {
+    /**
+     * Returns the name of parameter defined in customer function.
+     * The input is the simple class name of target annotation.
+     * @param annotationSimpleClassName - The simple class name of target annotation
+     * @return The name of parameter defined in customer function
+     */
+    Optional<String> getParameterName(String annotationSimpleClassName);

```

Comment:
if using `Optional` here, probably should use it for all public API that returns `null` (e.g. `getParameterValue` below)

my personal preference is generally using `@Nullable` annotation and not using `Optional`, unless writing a very functional-style API where you tend to do cool chaining with Optionals. we can chat more about pros/cons tomorrow.